### ISSUE TITLE
Uncaught (in promise) DOMException: play() can only be initiated by a user gesture.

### ISSUE THREAD

**Issue by Bahir** [richtr/NoSleep.js#19]
This message I receive on Android Chrome while do 
NoSleep.enable()

And NoSleep does not work

**Comment by GeertVanWonterghem** [comment]
Hi Bahir, I don't understand your question. Can you rephrase?

**Comment by jraoult** [comment]
For the record: "NoSleep" use a hack based on a looped video to keep the display awake. In Chrome for Android a video can only be started on user gesture.

It means that if you call `noSleep#enable` in anything else than a user dispatched event (click, touch etc.) you'll get this error.

**Comment by krembo** [comment]
Works for me only on 'click' event. None of touch events trigger NoSleep.

**Comment by OtterFamily** [comment]
yeah, they may have deprecated 'touchstart' and others as a valid initiator for audio/video playback because it bypasses the point of not just auto-playing since it's basically impossible to not fire a touch event when handling your mobile device.

**Comment by richtr** [comment]
As mentioned in the [README](https://github.com/richtr/NoSleep.js/blob/master/README.md), NoSleep.js `enable` must be wrapped in a browser event initiated by a user gesture:

> // (must be wrapped in a user input event handler e.g. a mouse or touch handler)

This is an acceptable compromise that means you can enable NoSleep if/when the user clicks on your page.

It was never the intention to support non-user-gesture sleep lock. Requiring a user gesture to enable sleep lock is an acceptable UX compromise for most experiences.

Closing this issue as this is not an issue with intended usage.
